File: app/src/main/java/com/cyberIyke/allvideodowloader/views/BadgeRed.java
package com.cyberIyke.allvideodowloader.views;

import android.content.Context;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatTextView;

public class BadgeRed extends AppCompatTextView {


    public BadgeRed(Context context) {
        super(context);
    }

    public BadgeRed(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public BadgeRed(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public void setNumber(int number) {
        if (number > 0) {
            show();
        } else {
            hide();
        }
        setText(String.valueOf(number));
    }

    public void hide() {
        setVisibility(GONE);
    }

    public void show() {
        setVisibility(VISIBLE);

    }


}

File: app/src/main/java/com/cyberIyke/allvideodowloader/views/BadgeRed.kt
package com.cyberIyke.allvideodowloader.views

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView

class BadgeRed : AppCompatTextView {
    constructor(context: Context?) : super(context!!) {}
    constructor(context: Context?, attrs: AttributeSet?) : super(
        context!!, attrs
    ) {
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context!!, attrs, defStyleAttr
    ) {
    }

    fun setNumber(number: Int) {
        if (number > 0) {
            show()
        } else {
            hide()
        }
        text = number.toString()
    }

    fun hide() {
        visibility = GONE
    }

    fun show() {
        visibility = VISIBLE
    }
}